net/pear-Auth_RADIUS/files/patch-Auth_RADIUS.php

```diff
@@ -12,53 +12,20 @@
      /**
       * Puts standard attributes.
       */ 
-@@ -300,6 +305,45 @@ class Auth_RADIUS {
-         $this->putAttribute(RADIUS_SERVICE_TYPE, RADIUS_FRAMED);
-         $this->putAttribute(RADIUS_FRAMED_PROTOCOL, RADIUS_PPP);
-         $this->putAttribute(RADIUS_CALLING_STATION_ID, isset($var['REMOTE_HOST']) ? $var['REMOTE_HOST'] : '127.0.0.1');
-+
-+		if (!function_exists("getNasIp")) {
-+			$ipaddr = "0.0.0.0";
-+		} else {
-+			$ipaddr = getNasIP();
-+		}
-+
-+		/*
-+		 * Add support for sending NAS-IP-Address, set this explicitly
-+		 * as an ip_addr
-+		 */
-+		$this->putAttribute(RADIUS_NAS_IP_ADDRESS, $ipaddr, "addr");
-+
-+		/* Add support for sending NAS-Identifier */
-+		if (function_exists("getNasID")) {
-+			$nasId = getNasID();
-+		} else if (!empty($cpzone) &&
-+		    empty($config["captiveportal"][$cpzone]["radiusnasid"])) {
-+			$nasId = php_uname("n");
-+		} else {
-+			$nasId =
-+			    $config["captiveportal"][$cpzone]["radiusnasid"];
-+		}
-+		$this->putAttribute(RADIUS_NAS_IDENTIFIER, $nasId);
-+
-+		if (function_exists("getNasPortType")) {
-+			$this->putAttribute(RADIUS_NAS_PORT_TYPE,
-+			    getNasPortType());
-+		}
-+
-+		if (function_exists("getNasPort")) {
-+			$this->putAttribute(RADIUS_NAS_PORT, getNasPort(),
-+			    'integer');
-+		}
-+
-+		if (function_exists("getCalledStationId")) {
-+			$this->putAttribute(RADIUS_CALLED_STATION_ID,
-+			    getCalledStationId());
-+		}
+@@ -295,11 +300,7 @@ class Auth_RADIUS {
+             $var = $GLOBALS['HTTP_SERVER_VARS'];
+         }
+                 
+-        $this->putAttribute(RADIUS_NAS_IDENTIFIER, isset($var['HTTP_HOST']) ? $var['HTTP_HOST'] : 'localhost');
+-        $this->putAttribute(RADIUS_NAS_PORT_TYPE, RADIUS_VIRTUAL);
+-        $this->putAttribute(RADIUS_SERVICE_TYPE, RADIUS_FRAMED);
+-        $this->putAttribute(RADIUS_FRAMED_PROTOCOL, RADIUS_PPP);
+-        $this->putAttribute(RADIUS_CALLING_STATION_ID, isset($var['REMOTE_HOST']) ? $var['REMOTE_HOST'] : '127.0.0.1');
++        $this->putAttribute(RADIUS_SERVICE_TYPE, RADIUS_LOGIN);
```

Comment:
Why are you removing attributes that are assigned by the original port? Less changes we need to carry on it are better.

Reply:
- For ```Service-Type``` and ```Framed-Protocol``` attributes : because these 2 attributes are wrong : they indicate that pfSense is querying Radius server in order to make a PPP connection. These two attributes were also not assigned in old ```RADIUS.inc```. 
- For ```Calling-Station-ID``` : The new RADIUS port broke this attribute : it is currently set to ```127.0.0.1```. I suggested in pfsense/pfsense#3640 to set this attribute [here in auth.inc](https://github.com/pfsense/pfsense/pull/3640/files#diff-647553ae2000651d4860da0cbf861caeR1651) _(for user manager authentication)_ and [here in captiveportal.inc](https://github.com/pfsense/pfsense/pull/3640/files#diff-764cc5453dc616c068ba7adeb68ba73eR2669) _(for captiveportal accounting)_, so i am removing it from the RADIUS class.